*** getchar.c	Sat May 13 03:57:57 2006
--- getchar.c.new	Sat May 13 03:57:57 2006
***************
*** 1695,1700 ****
--- 1695,1718 ----
  	}
  #endif
  
+ #ifdef FEAT_AUTOCMD
+ 	if (has_getchar_event())
+ 	{
+ 	    int c2;
+ 	    c2 = c;
+ 	    set_getchar_event_vvars( c );
+ 	    apply_autocmds(EVENT_GETCHAR, NULL, NULL, FALSE, curbuf);
+ 	    c = get_getchar_event_vvar_getchar();
+ 	    if ( c == NUL && c != c2 ) // && vgetorpeek(FALSE) != 0 )
+ 	    {
+ 		/* User has cleared the v:getchar variable */
+ 		old_char = -1;
+ 		continue;
+ 	    }
+ 	}
+ #endif
+ 
+ 
  	return c;
      }
  }
***************
*** 1711,1716 ****
--- 1729,1735 ----
      c = vgetc();
      if (c == NUL)
  	c = get_keystroke();
+ 
      return c;
  }
  
